reco_utils/nni/nni_utils.py

```diff
@@ -0,0 +1,58 @@
+import json
+import numpy as np
+import os
+import requests
+import time
+
+NNI_REST_ENDPOINT = 'http://localhost:8080/api/v1/nni'
+NNI_STATUS_URL = NNI_REST_ENDPOINT + '/check-status'
+NNI_TRIAL_JOBS_URL = NNI_REST_ENDPOINT + '/trial-jobs'
+
+
+def get_experiment_status(status_url):
+    nni_status = requests.get(status_url).json()
+    return nni_status['status']
+
+
+def check_experiment_status():
+    while True:
+        time.sleep(20)
+        status = get_experiment_status(NNI_STATUS_URL)
+        if status in ['DONE', 'TUNER_NO_MORE_TRIAL']:
+            break
+        elif status not in ['RUNNING', 'NO_MORE_TRIAL']:
+            raise RuntimeError("NNI experiment failed to complete with status {}".format(status))
+
+
+def check_stopped():
+    while True:
+        time.sleep(20)
+        try:
+            get_experiment_status(NNI_STATUS_URL)
+        except:
+            break
+
+
+def check_metrics_written():
+    while True:
+        time.sleep(20)
+        all_trials = requests.get(NNI_TRIAL_JOBS_URL).json()
+        if np.all(['finalMetricData' in trial for trial in all_trials]):
+            break
+
+
+def get_trials(optimize_mode):
+    if optimize_mode not in ['minimize', 'maximize']:
+        raise ValueError("optimize_mode should equal either 'minimize' or 'maximize'")
+    all_trials = requests.get(NNI_TRIAL_JOBS_URL).json()
```

Comment:
why is eval used here?

another option which is a little cleaner:
```
sorted_trials = sorted(all_trials, key=lambda x: x[0]['default'], reverse=optimize_mode=='maximize')
trials =  [(eval(trial['finalMetricData'][0]['data']), trial['logPath'].split(':')[-1]) for trial in sorted_trials]
best_trial_path = trials[0][1]
with open(os.path.join(best_trial_path, 
```

Reply:
`trial['finalMetricData'][0]['data']` is a string. `eval` here is doing the same thing as `json.loads()` and returns a dict.